docs/book/ahead-of-time-factories.md

```diff
@@ -0,0 +1,105 @@
+# Ahead of Time Factories 
+
+- Since 4.0.0
+
+In addition to the already existing [Reflection Factory](TODO), one can create factories for those services using `ReflectionBasedAbstractFactory` before deploying the project to production.
```

Comment:
@froschdesign How can I link from here to the documentation page of `Reflection Factory`?

Reply:
The same folder, then:

```markdown
[Reflection Factory](reflection-abstract-factory.md)
```

Reference: https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages